designs/containers/workload-discovery/workload-discovery.md

```diff
@@ -0,0 +1,185 @@
+### Kubernetes Version and Configuration
+* **Kubernetes Version:** 1.6.7. 
+* **Network Layer**:  [OVN](#ovn-kubernetes) with [SFC](#ovs-sfc) enabled or Nuage.
+* **Authentication**: For this first release the access to the Kubernetes API is allowed over `http` and **without user authentication**.  This should be acceptable as the intent of this release is for prototype/demonstration only rather than production readiness. For the longer run we want to allow only `https` access with authenticated users, this should be possible with [Fabric8](#fabric8-credential-example), the java SDK being adopted. 
+
+### OVN with SFC for Kubernetes
+> This assumption is being called out given that the traffic redirection using OVN SFC is still under investigation.
+
+To create a traffic redirection using OVN SFC additional calls to the OVN northbound database will be needed to retrieve the **switch name** and **port id** of the pod under protection. For that the following values from Kubernetes will suffice: **name of the node hosting the pod, pod namespace, pod name.**
+
+### Nuage SDN for Kubernetes
+> This assumption is being called out given that the traffic redirection using Nuage must still be tried out. 
+
+To create a traffic redirection using Nuage for Kubernetes calls to the Nuage SDN controller through the OSC Nuage plugin will be needed to retrieve the   **port id** of the pod under protection. The id of the port under protection is the only thing needed with respect to the workload pod to create the redirection. For that, the following values from Kubernetes will suffice: **name of the node hosting the pod, pod namespace, pod name.**  
+
+
+### Design Assumptions
+> Note: The assumptions below refer to design changes not yet fully described in this document but necessary to understand some of its current content. In the next revision of this document those details will be added (look for **TBD on the next revision** along this document) and these assumptions removed.  
+
+* **Security Group:**
+The security group entity and DTO in OSC will have a set of labels. These labels will be on the form `key=value`. For the first release of this feature all the `key` values in a given security group will always be the same, i.e.: security=webserver, security=backend, etc.
+
+* **Pod Security Group Member:**
+A new type of security group member is created, **Pod** along with the type **PodPort**. The **Pod** will store a `namespace`, `name`, `externalid`, `nodename` and a collection of `PodPort`.  The **PodPort** will store an `externalid`, a `collection of macaddresses` and a collection of `ipaddress`. 
+> Note: Although currently a pod can only have one ip address [support for multiple ip addresses and networks is intended](#k8s-pods-with-multiple-ips-issue).
+
+* **Virtualization Connector:**
+A new type of Virtualization Connector (VC) will be added: **Kubernetes**. This VC will require also a new field in the `providerAttributes` map: `providerPort`.

```

Comment:
is the provider port expected to be different for each installation....i would just use default and add in this functionality if needed